src/engine/strat_engine/pool.rs

```diff
@@ -211,24 +215,25 @@ impl StratPool {
         self.block_devs
             .save_state(&now().to_timespec(), data.as_bytes())
     }
-    /// Return an extend size for the physical space backing a pool
-    /// TODO: returning the current size will double the space provisoned to
-    /// back the pool.  We should determine if this is a reasonable value.
-    fn extend_size(&self, current_size: Sectors) -> Sectors {
-        current_size
-    }
 
-    /// Expand the physical space allocated to a pool by the value from extend_size()
-    /// Return tne number of Sectors added
+    /// Expand the physical space allocated to a pool.
+    /// Return tne number of Sectors added.
+    // TODO: Improve this function:
+    // * If there is enough space available the size of the new region is
+    // double the size of the old. This is most likely too greedy, we need a
+    // more sophisticated policy.
+    // * This method succeeds if the pool can be extended at all. This is
+    // most likely too generous a policy, as a very small extension may be
+    // no help at all.
```

Comment:
There's a granularity: if the thinpool DataBlock size is 4096, there is no benefit by extending the data device by 4095 sectors. Can we return a multiple of a given granularity?

But besides granularity, if we ask for 100 DataBlock equivalents and only get one, we'd certainly want to use that one, I'd think.